content/draft/general/keyboard_shortcuts.md

```diff
@@ -0,0 +1,111 @@
+---
+title: "Keyboard shortcuts spec"
+description: ""
+date: 2020-12-01T08:48:57+00:00
+lastmod: 2020-12-01T08:48:57+00:00
+draft: false
+images: []
+menu:
+  raw:
+    parent: "general"
+weight: 100
+toc: true
+---
+
+# Keyboard Shortcuts spec
+
+## Motivation
+
+Users often use keyboard shortcuts to more effectively navigate applications. This spec aims to document the keyboard shortcuts available as well as any preconditions.
```

Comment:
So generally speaking, this spec is not just about shortcuts but actually being able to navigate the application via the keyboard?

If that is the case then we should make this explicit in the document as well as the title of the spec.

Reply:
@PascalPrecht I think 'command' style shortcuts (e.g. CTRL+[a key] style shortcuts) and focus based interaction (e.g. bringing up a focus ring, moving the focus ring around with cursor keys, using ENTER to activate and ESC to go a level up or exit) should be treated as two separate features and should have separate specs.

The reason I'm thinking this is because 'command' style shortcuts are relatively straightforward to implement, but implementing spatial focus based interaction is much more complex.  Last time I worked on this (implementing spatial focus based interaction using QML) there were quite a few dev challenges to be overcome because QML doesn't have an inbuilt global spatial understanding e.g. if two different QML regions are next to each other, and a user presses a sideways cursor key, we would want the focus ring to jump to the item that's physically closest in the direction of the cursor key, and if this means jumping to a different region it's difficult to know the relative location of items in the other regions.